cli/azd/pkg/project/framework_service_dotnet_container_publish.go

```diff
@@ -0,0 +1,81 @@
+package project
+
+import (
+	"context"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/async"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+)
+
+type dotNetContainerPublishProject struct {
+	inner FrameworkService
```

Comment:
I'm having a hard time understanding the root reason we need this new framework service implementation? This is just a variant of a dotnet project that uses dotnet publish for image deployment??

It feels like this check could be handled in the `ContainerHelper` implementation?